---
title: Readcount analysis
author: Kevin Leiss
---

<!-- 
Removal of rRNA
Alignment pipeline
DeSeqU1: including changing the defaults, database used, and the gene lists
Cluster Viewer (also including updating)
-->


## _In silico_ rRNA removal
To save up disc space and speed up downstream analysis, reads of rRNA origin can be removed. 
The easiest and most straight forward approach is doing quality control using _FastQC_ and thereafter the removal of all overrepresented sequences which the tool identified. 
Another approach, which ensures only the removal of rRNA reads, is to provide a FASTA file of ribosomal sequences, which are then used to trim the sequenced reads.
In our developed pipeline _TrypRNAseq_, we use the tool _Cutadapt_ to do this task.

## Alignment pipeline
The following steps include the (I) alignment of the reads against the reference genome and (II) the counting of the reads per feature, i.e. gene, transcript, exon.
To streamline and ensure reproducibility, we developed the pipeline _TrypRNAseq_(see Note 1). 
As raw input for the pipeline the sequenced reads, a _Bowtie2_ indexed reference genome and its annotation (modified _GTF_ file) is needed.
<!-- How to generate the index and get the GTF file? -->


1. Retrieve the newest version of _TrypRNAseq_ from GitHub (https://github.com/agclayton/TrypRNAseq) (see Note 2).
2. Unpack the downloaded archive into a folder where you want to install the pipeline.
3. Place the read containing FASTQ files in a folder.
4. Start the pipeline by navigating within the terminal to the TrypRNAseq installation directory and enter the following into a terminal session: `python3 tryprnaseq.py`.

The interactive user guide will help the user to enter the desired set of parameters for alignment and readcount extraction. The standard parameter are set as follows: All overrepresented sequences are removed and all reads longer and equal to 30 bp in length are used for alignment. 
The pipeline will automatically run through the following steps:


1._Bowtie2_ aligns each read (see Note 3) against the indexed genome, up to 20 times, to account for gene duplicates. 
2. _samtools_ generates `.bam` files (see Note 4, `samtools view` command) of the alignments (`.sam`-format) and indexes them (`samtools index` command).
3. The read-counts for each gene are counted using a custom script. In short, the script uses the genome annotation (see Note 5) and counts the reads per listed feature using _samtools_. The output of this is saved in a tab-separated file.

### Updating the pipeline
<!-- 
1. Index genome
2. generate annotation GTF file
 -->

## Differential gene expression analysis
Differential expression (DE) analyses are used to identify differences between RNA-seq experiments.
The _R_-package _DEseq2_ is well established in the field and is able to model the gene expression based on two replicates, already.
We established a _R-shiny_ app called _DESeqUI_ to get a easy to use user-interface for _DEseq2_ and Trypanosomal RNA-seq experiments.
The app is designed to work seamingless with the output of the _TrypRNAseq_ pipeline, but can also be used to analyse Trypanosomal RNA-seq experiment readcounts, generated by custom pipelines.

1. Get the app from GitHub (https://github.com/agclayton/DESeqUI) and unpack the archive.
2. Open in _RStudio_ (ref) the file: `DESeqUI/app.R`
3. Click in _RStudio_ the "run app" button which will open the app in a browser window.

## Notes
1. We recommend to run the pipeline on a MacOS (the platform on which the pipeline was developed on) or a Linux machine. 
2. Make sure that the dependencies, which are listed on the GitHub page, are installed and available via the terminal, without the need to enter the full path of each tool: Add the paths to the different dependencies to your `$PATH` global variable before starting the pipeline. 
3. Only single-end sequnecing is possible for now.
4. `.bam`-files are only machine-readable but can be processed and indexed much faster than `.sam`-files.
